tests/sensors/test_external_task_sensor.py

```diff
@@ -214,6 +230,31 @@ def test_external_dag_sensor(self):
         )
         op.run(start_date=DEFAULT_DATE, end_date=DEFAULT_DATE, ignore_ti_state=True)
 
+    def test_external_dag_sensor_soft_fail_as_skipped(self):
+        other_dag = DAG('other_dag', default_args=self.args, end_date=DEFAULT_DATE, schedule_interval='@once')
+        other_dag.create_dagrun(
+            run_id='test', start_date=DEFAULT_DATE, execution_date=DEFAULT_DATE, state=State.SUCCESS
+        )
+        op = ExternalTaskSensor(
+            task_id='test_external_dag_sensor_check',
+            external_dag_id='other_dag',
+            external_task_id=None,
+            allowed_states=["failed"],
+            failed_states=["success"],
```

Comment:
```suggestion
            allowed_states=[State.Failed],
            failed_states=[State.Success],
```
WDYT about using the class attr rather than the harcoded value?

Reply:
Good call.